File: pset2/Vigenere/vigenere.c
//Encrypts according to the vigenere cipher

//Get libraries
#include <cs50.h> //get_string
#include <ctype.h> //isalpha isupper islower
#include <stdio.h> //printf
#include <string.h> //strlen

int main(int argc, string argv[])
{
    string key = argv[1]; //Getting the second element of the string array

    //There has to be 1 alphabetical command line argument
    if (argc == 2)
    {
        for (int i = 0; i < strlen(key); i++) //Iterate through the string
        {
            if (!(isalpha(key[i]))) //Check if each letter is not alphabetical
            {
                printf("Non grand baguette");
                return 1; //Return error
            }
        }
    }
    else //otherwise return error
    {
        printf("Non grand baguette");
        return 1;
    }

    string plaintext = get_string("plaintext: "); // Get plaintext
    printf("ciphertext: "); //Print ciphertext now - loops will print out a character at a time

    int j = 0; // counter - 'i' will increment with spaces and j will increment when a letter is enciphered

    for (int i = 0; i < strlen(plaintext); i++)
    {
        if (isupper(plaintext[i]) && isupper(key[j % strlen(key)])) // uppercase plaintext and uppercase key
        {
            printf("%c", (((plaintext[i] - ((int)'A')) + key[j % strlen(key)] - 'A') % 26) + ((int)'A'));
            j++;
        }
        else if (isupper(plaintext[i]) && islower(key[j % strlen(key)])) // uppercase plaintext and lowercase key
        {
            printf("%c", (((plaintext[i] - ((int)'A')) + key[j % strlen(key)] - 'a') % 26) + ((int)'A'));
            j++;
        }
        else if (islower(plaintext[i]) && islower(key[j % strlen(key)])) // lowercase plaintext and lowercase key
        {
            printf("%c", (((plaintext[i] - ((int)'a')) + key[j % strlen(key)] -  'a') % 26) + ((int)'a'));
            j++;
        }
        else if (islower(plaintext[i]) && isupper(key[j % strlen(key)])) // lowercase plaintext and uppercase key
        {
            printf("%c", (((plaintext[i] - ((int)'a')) + key[j % strlen(key)] - 'A') % 26) + ((int)'a'));
            j++;
        }
        else // if the letter isn't alphabetic then just print it out
        {
            printf("%c", plaintext[i]);
        } // Finishing this block increments i by 1

    }
    printf("\n");
    return 0;
}

File: pset6/vigenere/vigenere.py
from cs50 import get_string
import sys

def main():
    if len(sys.argv) == 2:
        key = sys.argv[1]
        for i in key:
            if not i.isalpha():
                print("Non you grand baguette")
                sys.exit(1)
                return
    else:
        print("Non you grand baguette")
        sys.exit(1)
        return
    
    plaintext = get_string("plaintext: ")
    print("ciphertext: ", end="")
    
    j = 0
    
    for i in plaintext:
        if i.isupper() and key[j % len(key)].isupper():
            print(chr((ord(i) - 65 + ord(key[j % len(key)]) - 65) % 26 + 65), end="")
            j+= 1
        
        elif i.isupper() and key[j % len(key)].islower():
            print(chr((ord(i) - 65 + ord(key[j % len(key)]) - 97) % 26 + 65), end="")
            j+= 1
        
        elif i.islower() and key[j % len(key)].islower():
            print(chr((ord(i) - 97 + ord(key[j % len(key)]) - 97) % 26 + 97), end="")
            j += 1
        
        elif i.islower() and key[j % len(key)].isupper():
            print(chr((ord(i) - 97 + ord(key[j % len(key)]) - 65) % 26 + 97), end="")
            j += 1
        
        else:
            print(i, end="")
        
    print()

if __name__ == "__main__":
    main()